src/http_proxy_io.c

```diff
@@ -918,6 +922,30 @@ static int http_proxy_io_set_option(CONCRETE_IO_HANDLE http_proxy_io, const char
                 result = 0;
             }
         }
+        else if (strcmp(option_name, OPTION_USE_TLS_IO) == 0)
+        {
+            (void)xio_close(http_proxy_io_instance->underlying_io, NULL, NULL);
+            xio_destroy(http_proxy_io_instance->underlying_io);
+            http_proxy_io_instance->use_tls_io = true;
+            const IO_INTERFACE_DESCRIPTION* underlying_io_interface;
+            underlying_io_interface = platform_get_default_tlsio();
+
+            TLSIO_CONFIG tls_io_config;
+            tls_io_config.hostname = http_proxy_io_instance->proxy_hostname;
+            tls_io_config.port = http_proxy_io_instance->proxy_port;
+            http_proxy_io_instance->underlying_io = xio_create(underlying_io_interface, &tls_io_config);
+
+            if (http_proxy_io_instance->underlying_io == NULL)
+            {
+                http_proxy_io_destroy(http_proxy_io);
```

Comment:
I don't think we should be destroying the handle. I don't think we own this here.

Reply:
I'm checking the socket_io creation logic in `http_proxy_io_create`, I find it would do some check and clean-up work.
do you think we also need to do the check -- tls interface and instance -- and do clean-up work in case of failure?

```c
if (result->underlying_io == NULL)
{
    /* Codes_SRS_HTTP_PROXY_IO_01_012: [ If xio_create fails, http_proxy_io_create shall fail and return NULL. ]*/
    LogError("Unable to create the underlying IO.");
    /* Codes_SRS_HTTP_PROXY_IO_01_008: [ When http_proxy_io_create fails, all allocated resources up to that point shall be freed. ]*/
    free(result->password);
    free(result->username);
    free(result->proxy_hostname);
    free(result->hostname);
    free(result);
    result = NULL;
}
```